jvcl/run/JvDriveCtrls.pas

```diff
@@ -1660,7 +1685,13 @@ procedure TJvFileListBox.SetDirectory(const Value: string);
      (AnsiCompareFileName(ExcludeTrailingPathDelimiter(FileName), ExcludeTrailingPathDelimiter(Directory)) <> 0) then
   begin
     inherited ApplyFilePath(Value);
-  end;
+    if assigned(FChangeNotify) and (FChangeNotify.Notifications.Count > 0) then
+      FChangeNotify.Notifications[0].Directory := Value;
+  end
+  else
+    // no directory defined any longer, so free change notification as well.
+    if assigned(FChangeNotify) then
+      FChangeNotify.Free;
```

Comment:
With this code `FChangeNotify` remains set to its previous value, which means any further `if Assigned` tests will return `True`. In turn, this will lead to crashes.
This should really be a call to `FreeAndNil(FChangeNotify)` and the remark about not needing the test in the line above is still valid.

Reply:
Implemented everything as requested. Please recheck and merge it in if good enough.